### PR TITLE
Rel Notes 4.11 Storage Content

### PR DESCRIPTION
**Pull request by lpettyjo**
**Version**(s): 4.11 only

**Issue**: Storage release notes content for 4.11

**Docs preview:** http://file.rdu.redhat.com/lpettyjo/rel_notes_4_11/release_notes/ocp-4-11-release-notes.html

**Additional information:**
Xrefs to new feature content not in place (??? placeholder) to avoid Travis check errors.

**PTAL**: @openshift/storage @jhou1

### CODE REVIEW

File: release_notes/ocp-4-11-release-notes.adoc [hunk 1]
```diff
@@ -94,6 +94,44 @@ For more information, see xref:../getting_started/openshift-overview.adoc[Gettin
...
+
+For more information, see ???
+
+[id="ocp-4-11-storage-csi-migration-cinder-ga"]
+==== Automatic CSI migration for OpenStack Cinder is generally available
+Starting with {product-title}  4.8, automatic migration for in-tree volume plug-ins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. Support for Cinder was provided in this feature in {product-title} 4.8, and {product-title} 4.11 now supports automatic migration for Cinder as generally available.
+
+This feature automatically translates in-tree objects to their counterpart CSI representations and should be completely transparent to users. Translated objects are not stored on disk, and user data is not migrated.
+
+While storage class referencing to the in-tree storage plug-in will continue working, we recommend that you switch the default storage class to the CSI storage class.
```
Comment (ahardin-rh) [inline release_notes/ocp-4-11-release-notes.adoc:109]: it is recommended

File: release_notes/ocp-4-11-release-notes.adoc [hunk 2]
```diff
@@ -94,6 +94,44 @@ For more information, see xref:../getting_started/openshift-overview.adoc[Gettin
...
+
+For more information, see ???
+
+[id="ocp-4-11-storage-csi-migration-azure-disk-ga"]
+==== Automatic CSI migration for Microsoft Azure Disk is generally available
+Starting with {product-title}  4.8, automatic migration for in-tree volume plug-ins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. Support for Azure Disk was provided in this feature in {product-title} 4.9, and {product-title} 4.11 now supports automatic migration for Azure File as generally available.
```
Comment (ahardin-rh) [inline release_notes/ocp-4-11-release-notes.adoc:115]: possible extra space before 4.8

File: release_notes/ocp-4-11-release-notes.adoc [hunk 3]
```diff
@@ -94,6 +94,44 @@ For more information, see xref:../getting_started/openshift-overview.adoc[Gettin
...
+
+This feature automatically translates in-tree objects to their counterpart CSI representations and should be completely transparent to users. Translated objects are not stored on disk, and user data is not migrated.
+
+While storage class referencing to the in-tree storage plug-in will continue working, we recommend that you switch the default storage class to the CSI storage class.
```
Comment (ahardin-rh) [inline release_notes/ocp-4-11-release-notes.adoc:119]: it is recommended

**Comment by ahardin-rh** [comment]
Just a few minor comments. Looks good!

File: release_notes/ocp-4-11-release-notes.adoc [hunk 4]
```diff
@@ -150,6 +150,44 @@ For more information, see link:https://kubernetes.io/docs/concepts/services-netw
...
 
+[id="ocp-4-11-storage-csi-azure-file-ga"]
+==== Persistent storage using Microsoft Azure File CSI Driver Operator is generally available
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Azure File. This feature was previously introduced as a Technology Preview feature in {product-title} 4.10 and is now generally available and enabled by default in {product-title} 4.11.
+
+For more information, see ???
+
+[id="ocp-4-11-storage-csi-migration-cinder-ga"]
+==== Automatic CSI migration for OpenStack Cinder is generally available
+Starting with {product-title}  4.8, automatic migration for in-tree volume plug-ins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. Support for Cinder was provided in this feature in {product-title} 4.8, and {product-title} 4.11 now supports automatic migration for Cinder as generally available.
```
Comment (gcharot) [inline release_notes/ocp-4-11-release-notes.adoc:161]: Should we mention that this is enabled by default? (no action required from the admin)

Comment (lpettyjo) [reply release_notes/ocp-4-11-release-notes.adoc:161]: Yes! I've added this.

Comment (gcharot) [reply release_notes/ocp-4-11-release-notes.adoc]: Thanks Lisa!

**Comment by jsafrane** [comment]
We should have a note that `snapshot.storage.k8s.io/v1beta1` API endpoint has been removed. Users should use `snapshot.storage.k8s.io/v1` that is available since 4.7. All objects created as v1beta1 are available through the v1 endpoint.

**Comment by lpettyjo** [comment]
> All objects created as v1beta1 are available through the v1 endpoint.

Done!

File: release_notes/ocp-4-11-release-notes.adoc [hunk 5]
```diff
@@ -212,6 +212,44 @@ For more information, see xref:../networking/external_dns_operator/nw-installing
...
+
+This feature automatically translates in-tree objects to their counterpart CSI representations and should be completely transparent to users. Translated objects are not stored on disk, and user data is not migrated.
+
+While storage class referencing to the in-tree storage plug-in will continue working, it is recommended that you switch the default storage class to the CSI storage class.
+
+For more information, see ???
+
+[id="ocp-4-11-storage-csi-migration-azure-disk-ga"]
+==== Automatic CSI migration for Microsoft Azure Disk is generally available
+Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. Support for Azure Disk was provided in this feature in {product-title} 4.9, and {product-title} 4.11 now supports automatic migration for Azure File as generally available. CSI migration for Azure Disk is now enabled by default and requires no action by an administrator.
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply release_notes/ocp-4-11-release-notes.adoc:233]: @lpettyjo please change Azure File in this sentence with Azure Disk.

Azure File CSI migration is still TP in 4.11.

**Comment by duanwei33** [comment]
@lpettyjo Hi I noticed in https://docs.openshift.com/container-platform/4.10/release_notes/ocp-4-10-release-notes.html#ocp-4-10-installation-vsphere-csi, there is a section `CSI driver installation on vSphere clusters` under `Installation and Upgrade` mentioned  vSphere/EXSi version, I think it should be changed to `7.0.2` as well, I'm not sure who will take that, just highlight here.
Others look good to me.

**Comment by lpettyjo** [comment]
> @lpettyjo Hi I noticed in https://docs.openshift.com/container-platform/4.10/release_notes/ocp-4-10-release-notes.html#ocp-4-10-installation-vsphere-csi, there is a section `CSI driver installation on vSphere clusters` under `Installation and Upgrade` mentioned vSphere/EXSi version, I think it should be changed to `7.0.2` as well, I'm not sure who will take that, just highlight here. Others look good to me.

@bscott-rh will handle this change in the install section.

**Comment by jsafrane** [comment]
Looks good to me.

File: release_notes/ocp-4-11-release-notes.adoc [hunk 6]
```diff
@@ -549,6 +549,62 @@ For clusters on user-provisioned xref:../installing/installing_bare_metal/instal
...
+==== Expanding CSI volumes is generally available
+Starting with {product-title} 4.3, expansion of Container Storage Interface (CSI) storage volumes after they have already been created became available as a Technology Preview feature, and is now generally available in {product-title} 4.11.
+
+For more information, see xref:../storage/expanding-persistent-volumes.adoc#expanding-csi-volumes_expanding-persistent-volumes[Expanding CSI volumes].
+
+[id="ocp-4-11-storage-csi-ephemeral_vols"]
+==== Support for CSI generic ephemeral volumes is generally available
+{product-title} 4.11 supports Container Storage Interface (CSI) generic ephemeral volumes as generally available. Generic ephemeral volumes are a type of ephemeral volume that can be provided by all storage drivers that also support persistent volumes and dynamic provisioning.
+
+For more information, see ???
```
Comment (dobsonj) [inline release_notes/ocp-4-11-release-notes.adoc:588]: I think this is the only remaining reference that needs to be updated, but the rest looks good to me.

Comment (lpettyjo) [reply release_notes/ocp-4-11-release-notes.adoc:588]: You are correct. After I merge Gen Ephemeral Vols PRs, then I will update the rel notes xref.

File: release_notes/ocp-4-11-release-notes.adoc [hunk 7]
```diff
@@ -1749,7 +1844,17 @@ In the table below, features are marked with the following statuses:
 |CSI Azure File Driver Operator
 |-
 |TP
-|TP
+|GA
+
+|CSI Azure Stack Hub Driver Operator
```
Comment (kalexand-rh) [inline release_notes/ocp-4-11-release-notes.adoc:1849]: @lpettyjo, some of these were intentionally removed on https://github.com/openshift/openshift-docs/pull/48581. Please do not reintroduce them.

Comment (lpettyjo) [reply release_notes/ocp-4-11-release-notes.adoc:1849]: @kalexand-rh I wasn't aware that we remove 3-GA items from the table (makes sense though). I've checked https://github.com/openshift/openshift-docs/pull/48581/ removals against my PR to ensure I wasn't adding back things that were removed.

**Pull request merged**